lib/enkf/ext_param.cpp

```diff
@@ -77,54 +90,115 @@ double ext_param_key_get( const ext_param_type * param, const char * key) {
   if (index < 0)
     util_abort("%s: invalid key:%s \n",__func__ , key);
 
-  return param->data[index];
+  return param->data[index].front();
+}
+
+
+double ext_param_key_suffix_get( const ext_param_type * param, const char * key, const char * suffix) {
+  int ikey = ext_param_config_get_key_index( param->config, key);
+  if (ikey < 0)
+    util_abort("%s: invalid key:%s \n",__func__ , key);
+
+  int isuffix = ext_param_config_ikey_get_suffix_index( param->config, ikey, suffix);
+  if (isuffix < 0)
+    util_abort("%s: invalid suffix:%s \n",__func__ , suffix);
+
+  return param->data[ikey][isuffix];
 }
 
 
 bool ext_param_iset( ext_param_type * param, int index , double value) {
-  if (index >= param->size)
+  if (index >= param->data.size())
     return false;
 
   if (index < 0)
     return false;
 
-  param->data[index] = value;
+  param->data[index].front() = value;
   return true;
 }
 
 
-const char* ext_param_iget_key(const ext_param_type * param, int index) {
-  return ext_param_config_iget_key( param->config , index );
+bool ext_param_iiset( ext_param_type * param, int ikey , int isuffix, double value) {
+  if (ikey >= param->data.size())
+    return false;
+
+  if (ikey < 0)
+    return false;
+
+  if (isuffix >= param->data[ikey].size())
+    return false;
+
+  if (isuffix < 0)
+    return false;
+
+  param->data[ikey][isuffix] = value;
+  return true;
 }
 
+
 double ext_param_iget(const ext_param_type * param, int index) {
-  if (index >= param->size)
-    util_abort("%s: invalid index:%d - range: [0,%d) \n",__func__ , index , param->size);
+  if (index >= param->data.size())
+    util_abort("%s: invalid index:%d - range: [0,%d) \n",__func__ , index , param->data.size());
 
   if (index < 0)
-    util_abort("%s: invalid index:%d - range: [0,%d) \n",__func__ , index , param->size);
+    util_abort("%s: invalid index:%d - range: [0,%d) \n",__func__ , index , param->data.size());
 
-  return param->data[index];
+  return param->data[index].front();
```

Comment:
I would expect `ext_param_iget` usage on a control with suffixes to be misuse. If you agree, should we raise instead of defaulting to first element for these methods?

Reply:
This was not intended as a default, this is the expected behavior in case the `ext_param` has no suffixes. But you are right, it can be misused.
I'd suggest to raise if this method is called but the `ext_param` has suffixes (and do the same in similar methods)

Reply:
No wait, we are in the C part, we cannot throw exception here. We could opt for a `util_abort`, but that is a bit extreme. I would prefer to do those checks only in python before calling this method, and in case raise a `RuntimeError`

Reply:
We could throw right? Because we are int the `C++` part ;)

I would support raising as you suggested both in the `C++` and Python code. Opting for the `C++` exceptions to never be raised via the Python bindings as long as the bindings do not propagate exceptions (Who knows, maybe pybind will drop by one day).

Reply:
I'm not entirely sure this would work. The implementation is now based on C++, but the API itself is pure C. Making an exception propagate outside of a public method would be rather inconsistent (unless we decide we are officially switching to C++ also in the API)

Also, I guess exceptions don't het along too well with cwrap, don't know what happens if an exception is thrown from a c++ piece of code that was called from python. I'll give it a try anyway and see what happens